src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -52,10 +52,19 @@ def to_matrix(x, dtype=np.double, copy=False):
       return x.copy("C"), True
     else:
       return x, False
+  elif (isinstance(x, np.ndarray) and x.dtype == dtype and x.flags.f_contiguous):
+    if copy: # Copy the matrix if required.
+      return np.ndarray(x.shape,buffer=x.flatten(),dtype=dtype,order = 'C').copy("C"),True
+    else:
+      return np.ndarray(x.shape,buffer=x.flatten(),dtype=dtype, order = 'C'),False
   else:
-    if isinstance(x, pd.core.series.Series):
-      x = pd.DataFrame(x)
-    return np.array(x, copy=True, dtype=dtype, order='C'), True
+    if isinstance(x, pd.core.series.Series) or isinstance(x, pd.DataFrame):
+      y = x.to_numpy()
```

Comment:
This only exists in Pandas 0.24 or newer, so we will need to handle cases where the Pandas version is older too.

Reply:
We can use `x.values`

Reply:
I think you are right, I did some playing with it and `x.values` does not seem to copy the matrix unless it is needed.